### PR TITLE
Version: better support for manual created latest and stable versions

### PR DESCRIPTION
**Pull request by stsewd**
Closes https://github.com/readthedocs/readthedocs.org/issues/11773

### CODE REVIEW

File: readthedocs/builds/models.py [hunk 1]
```diff
@@ -263,25 +262,20 @@ def external_version_name(self):
...
         version_name = self.verbose_name
-        if not self.is_external:
-            if self.slug == STABLE:
-                version_name = self.ref
-            elif self.slug == LATEST:
-                version_name = self.project.get_default_branch()
-            else:
-                version_name = self.slug
-            if "bitbucket" in self.project.repo:
-                version_name = self.identifier
```
Comment (stsewd) [inline readthedocs/builds/models.py:284]: There is nothing special about bitbucket, the branch/tag name is fine.

**Review by humitos — COMMENTED** [review]
I don't follow this change completely; but I think I don't fully understand the underlying issue either 🤷🏼 . I did a review, but I don't feel comfortable approving it because I'm confused. I will ask @ericholscher for a review here as well 😄

File: readthedocs/builds/models.py [hunk 2]
```diff
@@ -263,25 +262,20 @@ def external_version_name(self):
...
     @property
     def ref(self):
-        if self.slug == STABLE:
+        if self.slug == STABLE and self.machine:
             stable = determine_stable_version(
                 self.project.versions(manager=INTERNAL).all()
             )
             if stable:
-                return stable.slug
+                return stable.verbose_name
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/builds/models.py:270]: I didn't find any other references to this method, only on `vcs_url`.

Looks like you added that method p: https://github.com/readthedocs/readthedocs.org/pull/5356/files#r267728709

I think we can just move that code to live inside vcs_url, it doesn't need to be in a separate method.

File: readthedocs/builds/models.py [hunk 3]
```diff
@@ -263,25 +262,20 @@ def external_version_name(self):
...
     @property
     def vcs_url(self):
         version_name = self.verbose_name
-        if not self.is_external:
-            if self.slug == STABLE:
-                version_name = self.ref
-            elif self.slug == LATEST:
-                version_name = self.project.get_default_branch()
-            else:
-                version_name = self.slug
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/builds/models.py:282]: yeah, using the slug wasn't correct, as verbose_name has the actual name of the branch/tag.

**Comment by stsewd** [comment]
@humitos we have two types of latest and stable versions, the ones managed by rtd (machine=True) that basically are aliases, and the ones that the user manually creates (a tag/branch named latest/stable), in that case the branch/tag isn't an alias, but the branch/tag itself.

**Review by ericholscher — APPROVED** [review]
This change makes sense to me :+1:

**Pull request merged**